crates/router/src/connector/bitpay.rs

```diff
@@ -27,8 +31,18 @@ use crate::{
     utils::BytesExt,
 };
 
-#[derive(Debug, Clone)]
-pub struct Bitpay;
+#[derive(Clone)]
+pub struct Bitpay {
+    amount_converter: &'static (dyn AmountConvertor<Output = MinorUnit> + Sync),
```

Comment:
Why is this static lifetime

Reply:
Since we are passing this value as reference and want the value to live for the entire duration of the program, we have added `static` lifetime

Reply:
pub type BoxedConnector = Box<&'static (dyn Connector + Sync)>;

error[E0716]: temporary value dropped while borrowed
   --> crates/router/src/types/api.rs:475:53
    |
475 |                     Ok(ConnectorEnum::Old(Box::new(&connector::Stripe::new())))
    |                                                    -^^^^^^^^^^^^^^^^^^^^^^^^
    |                                                    ||
    |                                                    |creates a temporary value which is freed while still in use
    |                                                    cast requires that borrow lasts for `'static`
476 |                 }

without static lifetime we get the above error, and RPITIT can make this into static dispatch but we are using GAT which violate trait object safety. So we are using dynamic dispatch

Reply:
This doesnt have to dynamic dispatch, It could be static dispatch `'static dyn` is basically impl

Reply:
dyn keyword itself means dynamic dispatch, dyn can't be static dispatch.